### DISCUSSION TITLE
Aurora Engine ETH-JSON Test Requirements

### DISCUSSION THREAD

**Discussion by joshuajbouw** [aurora-is-near/aurora-engine#476] [Design]
The Ethereum JSON tests are a golden standard in terms of ensuring that non-geth applications are performing as expected. While [Sputnik EVM](https://todo.com) has incorporated the themselves, there are more than just the EVM tests we need to cover. For example; RLP, state, and also the same tests that [Sputnik EVM tests]()  covers.

There are two reasons why we require these tests:
1. Ensure full compatibility with ETH's EVM.
2. Full inclusion of the tests including Sputnik's will make integrating a potential different EVM backend easier to determine the correctness.

## Requirements

- The JSON per test must be parsed
- The tests must submit transactions to the Engine in an non-contract state, as well as through NEAR's testing utilities.
    - Do note that RLP tests are not via submitted transactions.
- Anything that is found to be deficient must be reported to @joshuajbouw immediatly.

## ETH JSON Tests Suites

The following are the required tests deemed to ensure that we are fully compliant.

### General State tests

https://github.com/ethereum/tests/tree/v10.3/GeneralStateTests

### RLP tests

https://github.com/ethereum/tests/tree/v10.3/RLPTests

### Gas limit tests

https://github.com/ethereum/tests/tree/v10.3/TransactionTests/ttGasLimit

### Transaction tests

https://github.com/ethereum/tests/tree/v10.3/TransactionTests

**Comment by andrcmdr** [comment]
I guess this related to our [previous discussion about testing framework](https://github.com/aurora-is-near/aurora-engine/discussions/377)?
